utils/build_profiler_report.py

```diff
@@ -0,0 +1,199 @@
+        if baseline_dict:
+            if target_mismatch:
+                if target in baseline_dict.keys():
+                    delta = current_dict[target]/total_time_current_intersect * 100 - \
+                        baseline_dict[target]/total_time_baseline_intesect * 100
+            else:
+                delta = perc - (baseline_dict[target]/total_time_baseline * 100)
+            if abs(delta) < 0.1:
+                delta = 0
+
+        # time is the formatted build time of the target
+        time = format_time(current_dict[target])
+
+        # time_delta is the formatted time difference between current and baseline
+        if baseline_dict and target in baseline_dict.keys():
+            time_delta = current_dict[target] - baseline_dict[target]
+            if abs(time_delta) < 60:
+                time_delta = 0
+            time_delta = format_time(time_delta)
+        else:
+            time_delta = 0
+
+        # perc_time_delta is a percentage difference of before and after build times
+        if baseline_dict and target in baseline_dict.keys():
+            perc_time_delta = (current_dict[target]/baseline_dict[target]) * 100 - 100
+        else:
+            perc_time_delta = 0
+
+        line = [f'{target:60}', f"{perc:4.1f}", f"{delta:5.1f}", f"{time:>8}",
+                f"{time_delta:>8}", f"{perc_time_delta:5.1f}"]
+        # if target was not in baseline, append note
+        if baseline_dict and target not in baseline_dict.keys():
+            line.append("Not in baseline")
+        print(' | '.join(line))
+
+    # Print time and % delta of the whole build time
+    if baseline_dict:
+        # total_perc_time_delta is the percentage change of build times between current and baseline
+        total_time_delta = total_time_current - total_time_baseline
+        if abs(total_time_delta) < 60:
+            total_time_delta = 0
+        total_time_delta = format_time(total_time_delta)
+        total_perc_time_delta = (total_time_current / total_time_baseline) * 100 - 100
+        line = ["-----\nTotal time:", format_time(total_time_current),
+                "| TD", f'{total_time_delta:>8}', "| %TD", f'{total_perc_time_delta:+5.1f}']
+        # if there are different targets in current and baseline log, add intersect deltas,
+        # which compare build times while omitting conficting build targets
+        if target_mismatch:
+            intersect_time_delta = total_time_current_intersect - total_time_baseline_intesect
+            if abs(intersect_time_delta) < 60:
+                intersect_time_delta = 0
+            intersect_time_delta = format_time(intersect_time_delta)
+            line.append(f'| intersect TD {intersect_time_delta:>8}')
+            intersect_perc_time_delta = (total_time_current_intersect /
+                                         total_time_baseline_intesect) * 100 - 100
+            line.append(f'| intersect %TD {intersect_perc_time_delta:+5.1f}')
+        print(' '.join(line))
```

Comment:
But, for some reason, now I can see everything twice, there are always 2 items that differ only by the name, but they obviously represent the same CMake target. For example

```
/home/jcerny/work/git/scap-security-guide/build/rhel8/collect-remediations-rhel8 |  6.6 |  -0.3 |    16.6s |    -1.4s |  -8.0
rhel8/collect-remediations-rhel8                             |  6.6 |  -0.3 |    16.6s |    -1.4s |  -8.0
```

Have you experienced this?

Reply:
I use cmake-3.22.1-5.fc35.x86_64.

Reply:
To address the comment above (For some reason I can't reply to it), my npm packages get installed in /usr/local/bin, so try adding that to your PATH.

Concerning the duplicate targets, the cmake version shouldn't matter, as the profiling works with ninja. My ninja version is 1.10.2, which is the latest according to https://github.com/ninja-build/ninja/releases. Try running ninja --version and try updating to the latest version.

Reply:
Can you point me to some guide/docs? I can see some webtreemap only in /home/jcerny/node_modules/webtreemap but nothing like the in /usr/local/bin.

Reply:
my ninja is ninja-build-1.10.2-6.fc35.x86_64

Reply:
I have chechekd and it's duplicate also in ninja log. So it doesn't happen for you, right? Why do you think that CMake shouldn't matter? CMake is the system that defines the build targets.